src/app/components/elements/panels/SelectedEventDetails.tsx

```diff
@@ -30,6 +30,32 @@ export const countStudentsAndTeachers = (eventBookings: EventBookingDTO[]) => {
   };
 };
 
+export const countGenders = (eventBookings: EventBookingDTO[]) => {
+  const genders = {
+    male: 0,
+    female: 0,
+    other: 0,
+    preferNotToSay: 0,
+    unknown: 0,
+  };
+
+  eventBookings.forEach((booking) => {
+    const gender = booking.userBooked?.gender;
+    const bookingStatus = booking.bookingStatus;
+
+    if (gender && bookingStatus) {
+      const validStatus = ["CONFIRMED", "ATTENDED"].includes(bookingStatus);
+
+      if (gender === "MALE" && validStatus) genders.male++;
```

Comment:
How come the valid status is only taken into account for male and female but not the rest?

Reply:
🙊🙊 oops!

Reply:
Rather than having `&& validStatus` in every condition, you can have an `if(validStatus)` and inside you can have a switch (instead of multiple if/else).

Reply:
This one I was on the fence about, I'd tried it as a switch statement but it more than doubles the number of lines so I thought if/else was more concise.  I've changed it to a switch now however 😄

Reply:
My personal recommendation is that if you have to choose between clarity and number of lines, go for clarity. The only time I think about number of lines being a problem is when I have a function or class or file that is too long, but that's only because it's a sign that the function/class/module is doing too much (has too many responsibilities). Having more lines in total in your project is not a bad (or good) thing.